### ISSUE TITLE
Undefined symbol: _RCTRegisterModule when trying to build app

### ISSUE THREAD

**Issue by jon-eubank** [mrousavy/react-native-tracking-transparency#18]
I am encountering the following error when attempting to build my React Native app after adding the react-native-tracking-transparency package --

<img width="655" alt="Screen Shot 2022-03-21 at 9 08 08 PM" src="https://user-images.githubusercontent.com/17089037/159387481-31086d6f-3063-4e74-9305-21f44d6a51cd.png">

<img width="505" alt="Screen Shot 2022-03-21 at 9 09 09 PM" src="https://user-images.githubusercontent.com/17089037/159387583-5881a17e-39cf-47c6-9ea5-102344e726b0.png">

I've been trying to solve this for awhile now and am sure it is probably something simple, but I can't find the solution and searching online hasn't yielded many answers.  How do I make my app build correctly while using this package?

**Steps followed:**

1) yarn add react-native-tracking-transparency
2) yarn install
2) cd ios
3) pod install

React version: 17.0.2
React Native version: 0.67.4
Xcode version: 13.2.1

**Comment by CorentinHeroux** [comment]
Same error, did you find a solution ?

**Comment by jde8aa** [comment]
> Same error, did you find a solution ?

Sorry, no real solution yet.  It is quite annoying, but occasionally the app will build correctly if I do some combination of pod repo update and pod install, combined with cleaning the build via Xcode.

**Comment by jon-eubank** [comment]
It seems that this issue is related to the same Xcode 12 build issue that several other packages have had, see this comment -- https://github.com/facebook/react-native/issues/29633#issuecomment-694187116

It can be fixed by updating the podspec file to depend on React-Core instead of React as seen in the linked comment.

We will have to create a PR to permanently fix this issue, but you can temporarily fix it yourself manually by editing the spec or using a package-patching tool.

**Comment by niloufarzahiri92** [comment]
I am trying to run my react-native project in Xcode(ios) and I am faced with this error (run in real iPhone) :

Link react_native_tracking_transparency (arm64) 0.2 seconds * Undefined symbol: _RCTRegisterModule**
My project can run on Android, but on iOS, it could not run. I should mention that neither can run in the simulator nor on a real iPhone, and the strange thing is that I can archive it and I can do it via test flight. When I install the test flight application it works well, but I can not run the project.

I uninstall **react_native_tracking_transparency *** to see if the error is related to this package but as soon as I uninstall that, the same error(Undefined symbol: _RCTRegisterModule) *happens but with another package (react_native_spinkit), and I continue to uninstall also SpinKit package and again the error happens with another package. Also, it could not run in the previous Xcode version (13), when I updated my Xcode (14.1) for the first time after updating Xcode the app could run in Xcode, but then again the error happened and could not run the app.

my system config: Mac mini,chip apple M1,macOS:Ventura 13

Xcode: 14.1

react-native-cli: 2.0.1 react-native: 0.68.1

I tried different solutions but none of them works for me. Thanks in advance for any help from you.
